azure-iot-device/samples/pnp/pnp_temp_controller_with_thermostats.py

```diff
@@ -92,16 +93,92 @@ def create_max_min_report_response(values):
 
 
 async def send_telemetry_from_temp_controller(device_client, telemetry_msg, component_name=None):
-    await pnp_methods.pnp_send_telemetry(
-        device_client, telemetry_msg, component_name
-    )  # only sends telemetry values that have changed
+    pnp_msg = pnp_helper_summer_refresh.create_telemetry(telemetry_msg, component_name)
+    await device_client.send_message(pnp_msg)
+    print("Sent message")
     await asyncio.sleep(8)
 
 
 #####################################################
+# COMMAND TASKS
+
+
+async def execute_command_listener(
+    device_client,
+    component_name=None,
+    method_name=None,
+    user_command_handler=None,
+    create_user_response_handler=None,
+):
+    """
+    Coroutine for executing listeners. These will listen for command requests.
+    They will take in a user provided handler and call the user provided handler
+    according to the command request received.
+    :param device_client: The device client
+    :param component_name: The name of the device like "sensor"
+    :param method_name: (optional) The specific method name to listen for. Eg could be "blink", "turnon" etc.
+    If not provided the listener will listen for all methods.
+    :param user_command_handler: (optional) The user provided handler that needs to be executed after receiving "command requests".
+    If not provided nothing will be executed on receiving command.
+    :param create_user_response_handler: (optional) The user provided handler that will create a response.
+    If not provided a generic response will be created.
+    :return:
+    """
+    while True:
+        if component_name and method_name:
+            command_name = component_name + "*" + method_name
+        elif method_name:
+            command_name = method_name
+        else:
+            command_name = None
+
+        command_request = await device_client.receive_method_request(command_name)
+        print("Command request received with payload")
+        print(command_request.payload)
+
+        # TODO should npt need this once the command request envelope is removed.
```

Comment:
I'm not sure if we need this comment, and I'm not sure what it means.

Reply:
this is the TODO I was referring to...